diff --git a/vm/vmcore/vm_topology/Cargo.toml b/vm/vmcore/vm_topology/Cargo.toml
index 27d5a964..1ddc587d 100644
--- a/vm/vmcore/vm_topology/Cargo.toml
+++ b/vm/vmcore/vm_topology/Cargo.toml
@@ -6,12 +6,16 @@ name = "vm_topology"
 edition.workspace = true
 rust-version.workspace = true
 
+[package.metadata.verus]
+verify = true
+
 [features]
 default = []
 inspect = ["dep:inspect", "memory_range/inspect"]
 mesh = ["dep:mesh_protobuf", "memory_range/mesh"]
 
 [dependencies]
+vstd.workspace = true
 aarch64defs.workspace = true
 cxl_spec.workspace = true
 memory_range.workspace = true
diff --git a/vm/vmcore/vm_topology/src/processor.rs b/vm/vmcore/vm_topology/src/processor.rs
index b6c70116..c7cba3d8 100644
--- a/vm/vmcore/vm_topology/src/processor.rs
+++ b/vm/vmcore/vm_topology/src/processor.rs
@@ -20,6 +20,10 @@
     }
 }
 use thiserror::Error;
+use vstd::prelude::*;
+
+#[cfg(verus_keep_ghost)]
+include!("processor.proof.rs");
 
 /// A description of the VM's processor topology.
 ///
@@ -259,10 +263,13 @@ pub struct VpTopologyInfo {
 /// them interchangeably.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
 #[cfg_attr(feature = "inspect", derive(inspect::Inspect), inspect(transparent))]
+#[verus_verify]
 pub struct VpIndex(u32);
 
 impl VpIndex {
     /// Returns `index` as a VP index.
+    #[verus_verify]
+    #[verus_spec(result => ensures result@ == index,)]
     pub const fn new(index: u32) -> Self {
         Self(index)
     }
@@ -277,6 +284,8 @@ pub const fn new(index: u32) -> Self {
     pub const BSP: Self = Self::new(0);
 
     /// Returns the VP index value.
+    #[verus_verify]
+    #[verus_spec(result => ensures result == self@,)]
     pub fn index(&self) -> u32 {
         self.0
     }
