--- a/vm/vmcore/src/vmtime.rs
+++ b/vm/vmcore/src/vmtime.rs
@@ -501,11 +501,12 @@
         self.reset_to(vmtime).await;
     }
 
+    #[verus_verify]
     async fn reset_to(&mut self, vmtime: VmTime) {
         assert!(!self.time.is_started(), "should be stopped");
         self.time = TimeState::Stopped(vmtime);
         self.req_send
-            .call(KeeperRequest::Reset, vmtime)
+            .call(|rpc| KeeperRequest::Reset(rpc), vmtime)
             .await
             .unwrap();
     }
